source/source.go

```diff
@@ -390,3 +393,8 @@ func isIPv6String(ip string) bool {
 	netIP := net.ParseIP(ip)
 	return netIP != nil && netIP.To4() == nil
 }
+
+func isIPv4String(input string) bool {
+	netIP := net.ParseIP(input)
+	return netIP != nil && netIP.To4() != nil
```

Comment:
Is this function intended to return `true` for IPv4-mapped IPv6 addresses?

Reply:
It just returns `true` when the input string is an IPv4 address. It'll return `false` if its IPv6 or otherwise invalid IP address.

Reply:
https://go.dev/play/p/2SNDvrwxdh-

Reply:
how about something like this?
```
func isIPv4String(input string) bool {
	netIP := net.ParseIP(input)
	return netIP != nil && netIP.To4() != nil && !strings.Contains(input, ":")
}
```